apps/evans/components/core/AudioPlayer.tsx

```diff
@@ -1,39 +1,34 @@
 import React, { useEffect, useRef, useState } from 'react';
 import cx from 'classnames';
-import { BackwardIcon, ForwardIcon } from '@heroicons/react/24/solid';
-import styles from '../../styles/AudioPlayer.module.css';
+import styles from '@/styles/AudioPlayer.module.css';
 import { formatDuration } from '@/lib/dates';
 
 interface Props {
   tracks: Array<{
     title: string;
-    url: string;
+    mp3Url: string;
+    duration: number;
   }>;
 }
 
 const AudioPlayer: React.FC<Props> = ({ tracks }) => {
   const [isPlaying, setIsPlaying] = useState(false);
   const [currentTime, setCurrentTime] = useState(0);
+  const [currentTrackIndex, setCurrentTrackIndex] = useState(0);
 
-  const currentTrackIndex = useRef(0);
   const audioRef = useRef<HTMLAudioElement>(null);
   const [audio, setAudio] = useState<HTMLAudioElement | null>(null);
+  const track = tracks[currentTrackIndex];
 
-  const currentTitle = tracks[currentTrackIndex.current]?.title ?? ``;
-
-  if (audio) {
-    if (audio.currentTime === audio.duration) {
-      if (currentTrackIndex.current === tracks.length - 1) {
-        setIsPlaying(false);
-        setCurrentTime(0);
-        audio.currentTime = 0;
-      } else {
-        currentTrackIndex.current++;
-        setCurrentTime(0);
-        setIsPlaying(true);
-        audio.currentTime = 0;
-      }
+  if (audio && audio.currentTime === audio.duration) {
```

Comment:
since the state `audio` and the `audioRef.current` point to the same object, i changed some things so we basically always deal with the ref through the state variable, as the source of truth (and b/c it's a little shorter)